src/schemapack/_internals/spec/datapack.py

```diff
@@ -227,3 +227,68 @@ def validate_root_resource(self) -> Self:
                 {"root_resource": self.rootResource, "root_class": self.rootClass},
             )
         return self
+
+    @model_validator(mode="after")
+    def validate_relation_classes(self) -> Self:
+        """Checks if the `targetClass` of relations defined in a datapack exists
+        within datapack resources.
+        """
+        non_found_classes: dict[str, str] = {}  # target_class -> relation_name
+
+        for resources in self.resources.values():
+            for resource in resources.values():
+                if not resource.relations:
+                    continue
+
+                for relation_name, relation in resource.relations.items():
+                    if relation.targetClass not in self.resources:
+                        non_found_classes[relation.targetClass] = relation_name
+
+        if non_found_classes:
+            raise PydanticCustomError(
+                "TargetClassNotFoundError",
+                "Did not find the target class if the following relations (relation"
+                + " names): {non_found_classes}",
+                {
+                    "non_found_target_ids": ", ".join(
+                        f"'{target_class}' ('{relation_name}')"
+                        for target_class, relation_name in non_found_classes.items()
```

Comment:
How does string substitution work here, i.e. what does the final string look like?

Reply:
There was an error here. For the corrected version:

e.g. for the file `examples/datapack/invalid/simple_relations/DataPackSpecError.TargetClassNotFoundError.datapack.yaml`

This is the output:
```
The provided object was not compatible with the datapack specification: 1 validation error for DataPack                                                                                                                           Did not find the target class of the following relations (relation names): 'File' ('files') [type=TargetClassNotFoundError, input_value={'datapack': '0.3.0', 're...rgetResources': []}}}}}}, input_type=CommentedMap] 

```